domain.go

```diff
@@ -0,0 +1,62 @@
+package generictypes
+
+import (
+	"encoding/json"
+
+	"github.com/alexanderritola/validate"
+	"github.com/alexanderritola/validate/web"
+	"github.com/giantswarm/generic-types-go"
+	"github.com/juju/errgo"
+)
+
+type Domain string
+
+type DomainConfig map[Domain]generictypes.DockerPort
```

Comment:
`DomainConfig` is very `user-config` library specific. I would leave that over there.